src/main/java/org/usfirst/frc/team1806/robot/OI.java

```diff
@@ -49,59 +45,111 @@ public void runCommands(){
 			}
 		}
-		synchronized (mLiftSubsystem) {
-			if(dc.getButtonA()) {
-				mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.TELEOP_HOLD);
-			}
-			if(dc.getButtonX()) {
-				mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.SHIP_CARGO);
-			}
-			if(dc.getButtonY()) {
-				mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.ROCKET_CARGO_HIGH);
-			}
-		}
 
-		if(Constants.enableAutoInTeleOp){
-			autoInTeleOp.update(autoController.getButtonStart());
-		}
+		//Controls that change based on mode
+		switch(Robot.getGamePieceMode()){
+			case HATCH_PANEL:
+			default:
 
-		if(!wasB && dc.getButtonB()){
-			if(mSquidSubsystem.isOpen()) {
-				mSquidSubsystem.closeSquid();
-			}
-			else {
-				mSquidSubsystem.openSquid();
-			}
-		}
+				synchronized (mLiftSubsystem) {
+					if(dc.getButtonA()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.ROCKET_HATCH_LOW);
+					}
+					if(dc.getButtonX()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.ROCKET_HATCH_MID);
+					}
+					if(dc.getButtonY()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.ROCKET_HATCH_HIGH);
+					}
+					if(dc.getButtonB()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.BOTTOM_LIMIT);
+					}
+				}
 
-		if(!wasRightDPad && dc.getPOVRight()) {
-			if(mSquidSubsystem.isExtended()) {
-				mSquidSubsystem.retractSquid();
-			}
-			else {
-				mSquidSubsystem.isExtended();
-			}
-		}
-		if(dc.getLeftTrigger() >.2){
-			mCargoIntakeSubsystem.intakeCargo();
-		}
-		else if(dc.getPOVDown()){
-			mCargoIntakeSubsystem.scoreCargo(CargoIntakeSubsystem.ScoringPower.SLOW);
-		}
-		else if(dc.getPOVLeft()){
-			mCargoIntakeSubsystem.scoreCargo(CargoIntakeSubsystem.ScoringPower.FAST);
-		}
-		else if(dc.getPOVUp()){
-			mCargoIntakeSubsystem.scoreCargo(CargoIntakeSubsystem.ScoringPower.IRRESPONSIBLE);
+				synchronized (mSquidSubsystem) {
+					if (dc.getLeftTrigger() > Constants.kTiggerThreshold) {
+						mSquidSubsystem.openSquid();
+					}
+
+					if (dc.getButtonLB()) {
+						mSquidSubsystem.closeSquid();
+					}
+
+
+					if (!wasSquidExtendButton && dc.getPOVRight()) {
+						if (mSquidSubsystem.isExtended()) {
+							mSquidSubsystem.retractSquid();
+						} else {
+							mSquidSubsystem.isExtended();
+						}
+					}
+				}
+
+
+				//TODO:Beaver controls
+
+
+				if(dc.getButtonRB() && !wasChangeModeButton){
+					Robot.setGamePieceMode(Robot.GamePieceMode.CARGO);
+				}
+
+				break;
+
+			case CARGO:
+
+				synchronized (mLiftSubsystem) {
+					if(dc.getButtonA()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.ROCKET_CARGO_LOW);
+					}
+					if(dc.getButtonX()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.ROCKET_CARGO_MID);
+					}
+					if(dc.getButtonY()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.ROCKET_CARGO_HIGH);
+					}
+					if(dc.getButtonB()) {
+						mLiftSubsystem.goToSetpoint(LiftSubsystem.LiftPosition.BOTTOM_LIMIT);
```

Comment:
Bringing the lift to the bottom is probably more common than switching modes... should we put that on RB and make B the switch mode button?

Reply:
I think we probably want most lift setpoints in the same area on the controller if possible, and having an accessible button for the switch mode will be easier to remember. It's all up to Joel though.